libcodechecker/libhandlers/server.py

```diff
@@ -470,18 +766,55 @@ def main(args):
         prod_server = database.SQLiteDatabase(
             default_product_path, RUN_META,
             context.run_migration_root, check_env)
-        prod_server.connect(context.run_db_version_info, init=True)
 
-        LOG.debug("Connecting database engine for default product")
+        LOG.debug("Checking 'Default' product database.")
+        db_status = prod_server.connect()
+        if db_status != DBStatus.MISSING:
+            db_status = prod_server.connect(init=True)
+            LOG.error(database_status.db_status_msg.get(db_status))
+            if db_status != DBStatus.OK:
+                LOG.error("Failed to configure default product")
+                sys.exit(1)
+
         product_conn_string = prod_server.get_connection_string()
-        LOG.debug("Default database created and connected.")
 
         server.add_initial_run_database(
-            sql_server, product_conn_string)
+            cfg_sql_server, product_conn_string)
 
         LOG.info("Product 'Default' at '{0}' created and set up."
                  .format(default_product_path))
 
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+
+    upgrade_available = {}
+    for k, v in prod_statuses.items():
+        db_status, _, _, _ = v
+        if db_status == DBStatus.SCHEMA_MISMATCH_OK or \
+                db_status == DBStatus.SCHEMA_MISSING:
+            upgrade_available[k] = v
+
+    if upgrade_available:
+        print_prod_status(prod_statuses)
+        LOG.warning("Products can be upgraded take a backup!")
```

Comment:
And please, change this line. `Databases can be upgraded, please make a backup!` (Products??) (And `take` a backup doesn't sound right!)